### PR TITLE
added rain water problem code in Cpp, C and python 

### PR DESCRIPTION
**Pull request by divs30**
Added rain water problem code in Cpp, C and  Python 

Fixes # Issue number #894 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added implementation of rain water trapping problem (codes) in C,C++ and Python with comments and code.

#### Languages Used:
- C++
- C
- Python

#### Files Added:
- Rain_Water_Trapping/Rain_Water_Trapping.c
- Rain_Water_Trapping./Rain_Water_Trapping.cpp
- Rain_Water_Trapping./Rain_Water_Trapping.py

### CODE REVIEW

**Comment by gupta2140** [comment]
Please fix the template and fill it properly.

**Comment by tsah20** [comment]
Please remove the docx file in the PR

**Comment by divs30** [comment]
removed doc file and added python code

File: Rain_Water/Rain_Water_trapping.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	static int WaterCalculator(int size) 
+	{ 
+		// left[i] stores ht of tallest bar to the lf of the i'th bar including itself 
+		 
+		int left[] = new int[size]; 
+	
+		// Right [i] stores ht of tallest bars to the rt of ith bar including itself 
+		// 
+		int right[] = new int[size]; 
+
```
Comment (tsah20) [inline Rain_Water/Rain_Water_trapping.java]: Remove the spaces

File: Rain_Water/Rain_Water_trapping.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	
+	// Driver method to test the above function 
+	public static void main(String[] args) 
+	{ 
+		
+		System.out.println("Maximum collected water  is " + 
+										WaterCalculator(a.length)); 
+	} 
+} 
+//output Maximum collected water is 6
```
Comment (tsah20) [inline Rain_Water/Rain_Water_trapping.java]: Add a new line at the EOF

**Comment by divs30** [comment]
resolved the suggested changes

**Comment by jainaman224** [comment]
Hey @divs30,
You file was correct, you only to fix file extension. Sorry, for not being clear.
Please use `Rain_Water_Trapping` for file name and `Rain_Water_Trapping` for file name. Add extension to files appropriately.

**Comment by jainaman224** [comment]
Please fix the template and fill it properly.

File: README.md. [hunk 3]
```diff
@@ -0,0 +1,12 @@
+Rain water trapping problem
```
Comment (SakshiShreya) [inline README.md.]: Change this line to heading. Read about markdown styling.

File: Rain_Water_Trapping/Rain_Water_Trapping.c [hunk 4]
```diff
@@ -0,0 +1,65 @@
+#include <stdio.h> 
+
+using namespace std;
+
+//function to calculate min
```
Comment (SakshiShreya) [inline Rain_Water_Trapping/Rain_Water_Trapping.c]: Add a space after `//` in all comments

File: Rain_Water_Trapping/Rain_Water_Trapping.c [hunk 5]
```diff
@@ -0,0 +1,65 @@
+#include <stdio.h> 
+
+using namespace std;
+
+//function to calculate min
+int min(int a, int b)
+{
+    return a < b ? a :b;
```
Comment (SakshiShreya) [inline Rain_Water_Trapping/Rain_Water_Trapping.c]: Add a space after `:`

File: Rain_Water_Trapping/Rain_Water_Trapping.c [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+int min(int a, int b)
+{
+    return a < b ? a :b;
+}
+
+
+//function to calculate max
+int max(int a, int b)
+{
+    return a > b ? a :b;
```
Comment (SakshiShreya) [inline Rain_Water_Trapping/Rain_Water_Trapping.c]: Add a space after `:`

File: Rain_Water_Trapping/Rain_Water_Trapping.c [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+
```
Comment (SakshiShreya) [inline Rain_Water_Trapping/Rain_Water_Trapping.c]: Remove extra lines between functions. Only one blank line should be present between two functions.

File: Rain_Water_Trapping/Rain_Water_Trapping.cpp [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+       water_volume += min(left[i], right[i]) - a[i]; 
+  
+    return water_volume; 
+} 
+  
+// Driver program 
+int main() 
+{ 
+    int n, i;
+    cin>>n; //taking input for the size of array
```
Comment (SakshiShreya) [inline Rain_Water_Trapping/Rain_Water_Trapping.cpp]: Add space around `>>`

File: Rain_Water_Trapping/Rain_Water_Trapping.cpp [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    int a[n];
+    for (i=0; i<n; i++)
```
Comment (SakshiShreya) [inline Rain_Water_Trapping/Rain_Water_Trapping.cpp]: Add space around all operators.

**Comment by divs30** [comment]
I was about to do modification today,may I do it,is there any scope to reopen this pr

File: README.md. [hunk 10]
```diff
@@ -1,4 +1,4 @@
-Rain water trapping problem
+# Rain water trapping problem
```
Comment (SakshiShreya) [inline README.md.]: Remove the last `.` in the file name.

~README.md.~
README.md

File: Rain_Water_Trapping/Rain_Water_Trapping.c [hunk 11]
```diff
@@ -1,42 +1,38 @@
+//Rain_Water_Trapping
```
Comment (SakshiShreya) [inline Rain_Water_Trapping/Rain_Water_Trapping.c]: Add a space after `//`

File: Rain_Water_Trapping/Rain_Water_Trapping.c [hunk 12]
```diff
@@ -1,42 +1,38 @@
...
-    return a < b ? a :b;
+    return a < b ? a : b;
 }
 
-
-//function to calculate max
+// function to calculate max
 int max(int a, int b)
 {
     return a > b ? a :b;
```
Comment (SakshiShreya) [inline Rain_Water_Trapping/Rain_Water_Trapping.c]: Add a space after `:`

File: Rain_Water_Trapping/Rain_Water_Trapping.cpp [hunk 13]
```diff
@@ -1,29 +1,28 @@
-#include<bits/stdc++.h> 
+//Rain_Water_Trapping
```
Comment (SakshiShreya) [inline Rain_Water_Trapping/Rain_Water_Trapping.cpp]: Add a space after `//`

File: Rain_Water_Trapping/Rain_Water_Trapping.py [hunk 14]
```diff
@@ -1,11 +1,11 @@
- #trapping rain water problem code in pyhton 
-def trappedWater(a, size): 
+ #Rain_Water_Trapping
```
Comment (SakshiShreya) [inline Rain_Water_Trapping/Rain_Water_Trapping.py]: Add a space after `#`

File: Rain_Water_Trapping/Rain_Water_Trapping.py [hunk 15]
```diff
@@ -16,24 +16,21 @@ def trappedWater(a, size):
...
-  
+    
+# main program   
 a = [0, 1, 0, 2, 1, 0, 1, 3, 2, 1, 2, 1] 
 size = len(a) 
-print("Maximum water that can be accumulated is ",trappedWater(a, size)) 
+print("Maximum water that can be accumulated is ", trappedWater(a, size)) 
   
-
 # output :Maximum water that can be accumulated is 6
```
Comment (SakshiShreya) [inline Rain_Water_Trapping/Rain_Water_Trapping.py]: Remove space before `:` and add a space after the same.

File: README.md. (hunk 10 above)
Comment (jainaman224) [reply README.md.]: It is not done.

**Pull request merged**